### PR TITLE
Tristan's README for JS Chats

### PR DESCRIPTION
**Pull request by GalenWong**

### CODE REVIEW

File: posts/spring2021/js-chats-6-hooks/index.md [hunk 1]
```diff
@@ -407,31 +406,33 @@ Yay! It works!
...
 
-In previous discussion, we mentioned reducer is also a concept in Redux. In fact, Redux is an implementation of Facebook's flux architecture.
-
-Imagine you have a project with multiple React components. For these components, they sometimes need to interact with each other by passing props and states. Without redux, you can only have a top-down data passing from root component to each of the component that requires a parent state, and the entire graph forms a directed acyclic graph. However, with the introduction of the flux architecture, all the components' states can be centralized to a structure called "store":
+The problem without an ultimate state manager looks like the following graph. In order to let child components to use the state from the root component (assume the blue circle is the root component), we need to perform a DFS-like graph traversal in order to let the state be used. This is inefficient in itself, but what makes matters worse is that if you store every state in the root component if one state has changed, every component needs to be re-rendered since every node is essentially a child of the root component, this is very inefficient.
```
Comment (Tristan816la) [inline posts/spring2021/js-chats-6-hooks/index.md:423]: Can we change the start sentence to "The problem without an ultimate state manager can be illustrated by the following graph."

File: posts/spring2021/js-chats-6-hooks/index.md [hunk 2]
```diff
@@ -267,9 +266,9 @@ In fact, there is nothing special when you want to create your own hook. However
 
 By using custom hooks, we can interact with React states in really creative ways. However, when we talk about states in functional components, `useState` is not the only way that you can interact with states.
 
-### New Problem - Sign Up Form
+### New Problem - Sign-Up Form
 
-Suppose we're building a sign up form on the client side. For this form, we require the user to enter username, password (& confirm password) and email, and the user could submit the form or clear the form. You might be familiar with how to implement a form using useState(), so you came up with the following implementation (We ignored confirm password check on the client side):
+Suppose we're building a sign-up form on the client side. For this form, we require the user to enter user name, password (& confirm password), and email, and the user could submit the form or clear the form. You might be familiar with how to implement a form using useState(), so you came up with the following implementation (We ignored confirm password check on the client side):
```
Comment (Tristan816la) [inline posts/spring2021/js-chats-6-hooks/index.md:271]: Can we add a "the" before "password check" in the last sentence in parenthese?

File: posts/spring2021/js-chats-6-hooks/index.md (hunk 1 above)
Comment (GalenWong) [reply posts/spring2021/js-chats-6-hooks/index.md]: done

Comment (Tristan816la) [reply posts/spring2021/js-chats-6-hooks/index.md]: And the following sentence should be: "let child components use the state from the root component"(without "to"). I should really create a counter for my grammar mistakes.

**Comment by Tristan816la** [comment]
Sorry for all these tedious changes, but I also noticed two consistency issues. 1. I have used "React.FC" somewhere in the text and I think I should change all of them to "the function components" (there is still one React.FC in the redux section).  2. Some of the functions have no trailing "()" . I'll change all of them and send you the latest README.

**Pull request merged**